### PR TITLE
Perform conversion between colour types

### PR DESCRIPTION
**Pull request by TurkeyMan**

### CODE REVIEW

File: src/wg/color/rgb/package.d [hunk 1]
```diff
@@ -119,12 +135,178 @@ unittest
...
+//
+//    alias sRGBA = RGB!("rgba", ubyte, false, RGBColorSpace.sRGB);
+//
+//    // test linear conversion
+//    alias lRGBA = RGB!("rgba", ushort, true, RGBColorSpace.sRGB);
+//    assert(convertColorImpl!(lRGBA)(sRGBA(0xFF, 0xFF, 0xFF, 0xFF)) == lRGBA(0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF));
+//
+//    // test gamma conversion
+//    alias gRGBA = RGB!("rgba", byte, false, RGBColorSpace.sRGB_Gamma2_2);
+//    assert(convertColorImpl!(gRGBA)(sRGBA(0xFF, 0x80, 0x01, 0xFF)) == gRGBA(0x7F, 0x3F, 0x03, 0x7F));
```
Comment (drug007) [inline src/wg/color/rgb/package.d]: is unittest disabling intended?

**Comment by drug007** [comment]
Is there a reason to use own linalgebra module instead of some external package?

File: src/wg/color/rgb/package.d (hunk 1 above)
Comment (TurkeyMan) [reply src/wg/color/rgb/package.d]: I just need to rework the test. I reworked most of this code to suit this lib from my other color lib from years ago. We've done a lot of expressions differently here, but principles are all the same.

**Comment by TurkeyMan** [comment]
Because I only use 2 functions; 3x3 multiply and inverse. They're small trivial functions.
Dragging in a submodule is a big commitment, and doing so for 2 of the simplest functions is overkill.
This isn't a big commitment. It's easier if the repo is self-contained, and we can swap it out in the future if we see an advantage, or if this file grows unreasonably (I'm not aware of anything else we'll need that's not already in there).

If UI rendering emerges in *this* repo (I suspect UI might be a sibling repo because it'll likely need numerous dependencies), then we might need 4x4 maths, and at that point I'd consider a 3rd party lib. Let's just see how it develops?

**Comment by drug007** [comment]
Well, what I can see is you are developing it and I like it. Thank you!

**Comment by TurkeyMan** [comment]
I fixed the tests.

**Comment by p0nce** [comment]
There is tension between too little reuse and too many, and sometimes duplication is the right way.

**Pull request merged**